cmd/options/options.go

```diff
@@ -94,30 +97,38 @@ func NewNodeProblemDetectorOptions() *NodeProblemDetectorOptions {
-	fs.BoolVar(&npdo.EnableK8sExporter, "enable-k8s-exporter", true, "Enables reporting to Kubernetes API server.")
-	fs.StringVar(&npdo.ApiServerOverride, "apiserver-override",
-		"", "Custom URI used to connect to Kubernetes ApiServer. This is ignored if --enable-k8s-exporter is false.")
-	fs.DurationVar(&npdo.APIServerWaitTimeout, "apiserver-wait-timeout", time.Duration(5)*time.Minute, "The timeout on waiting for kube-apiserver to be ready. This is ignored if --enable-k8s-exporter is false.")
-	fs.DurationVar(&npdo.APIServerWaitInterval, "apiserver-wait-interval", time.Duration(5)*time.Second, "The interval between the checks on the readiness of kube-apiserver. This is ignored if --enable-k8s-exporter is false.")
-	fs.BoolVar(&npdo.PrintVersion, "version", false, "Print version information and quit")
-	fs.StringVar(&npdo.HostnameOverride, "hostname-override",
-		"", "Custom node name used to override hostname")
-	fs.IntVar(&npdo.ServerPort, "port",
-		20256, "The port to bind the node problem detector server. Use 0 to disable.")
-	fs.StringVar(&npdo.ServerAddress, "address",
-		"127.0.0.1", "The address to bind the node problem detector server.")
-
-	fs.IntVar(&npdo.PrometheusServerPort, "prometheus-port",
-		20257, "The port to bind the Prometheus scrape endpoint. Prometheus exporter is enabled by default at port 20257. Use 0 to disable.")
-	fs.StringVar(&npdo.PrometheusServerAddress, "prometheus-address",
-		"127.0.0.1", "The address to bind the Prometheus scrape endpoint.")
 
+	fs.BoolVar(&npdo.EnableK8sExporter, "enable-k8s-exporter", true, "Enables reporting to Kubernetes API server.")
+	fs.StringVar(&npdo.ApiServerOverride, "apiserver-override", "",
+		"Custom URI used to connect to Kubernetes ApiServer. This is ignored if --enable-k8s-exporter is false.")
+	fs.DurationVar(&npdo.APIServerWaitTimeout, "apiserver-wait-timeout", time.Duration(5)*time.Minute,
+		"The timeout on waiting for kube-apiserver to be ready. This is ignored if --enable-k8s-exporter is false.")
+	fs.DurationVar(&npdo.APIServerWaitInterval, "apiserver-wait-interval", time.Duration(5)*time.Second,
+		"The interval between the checks on the readiness of kube-apiserver. This is ignored if --enable-k8s-exporter is false.")
+	fs.IntVar(&npdo.ServerPort, "port", 20256,
+		"The port to bind the node problem detector server. Use 0 to disable.")
+	fs.StringVar(&npdo.ServerAddress, "address", "127.0.0.1",
+		"The address to bind the node problem detector server.")
+
+	fs.IntVar(&npdo.PrometheusServerPort, "prometheus-port", 20257,
+		"The port to bind the Prometheus scrape endpoint. Prometheus exporter is enabled by default at port 20257. Use 0 to disable.")
+	fs.StringVar(&npdo.PrometheusServerAddress, "prometheus-address", "127.0.0.1",
+		"The address to bind the Prometheus scrape endpoint.")
+
+	for _, exporterName := range exporters.GetExporterNames() {
+		exporterHandler := exporters.GetExporterHandlerOrDie(exporterName)
+		exporterHandler.Options.SetFlags(fs)
```

Comment:
Can you move the content of `SetFlags` here? It is harder to read the code to understand what flag is added in different places.

Reply:
Actually I created the `Options.SetFlags()` abstraction to allow each exporter plugin to define their own flags flexibly.

This is for #345 , and later I will send a PR to create `ExporterHandler` for k8s_exporter and prometheus_exporter. And in their `SetFlags()` function, existing options we have today will be added:
```
	fs.BoolVar(&npdo.EnableK8sExporter, "enable-k8s-exporter", true, "Enables reporting to Kubernetes API server.")
	fs.StringVar(&npdo.ApiServerOverride, "apiserver-override", "",
		"Custom URI used to connect to Kubernetes ApiServer. This is ignored if --enable-k8s-exporter is false.")
	fs.DurationVar(&npdo.APIServerWaitTimeout, "apiserver-wait-timeout", time.Duration(5)*time.Minute,
		"The timeout on waiting for kube-apiserver to be ready. This is ignored if --enable-k8s-exporter is false.")
	fs.DurationVar(&npdo.APIServerWaitInterval, "apiserver-wait-interval", time.Duration(5)*time.Second,
		"The interval between the checks on the readiness of kube-apiserver. This is ignored if --enable-k8s-exporter is false.")
	fs.IntVar(&npdo.ServerPort, "port", 20256,
		"The port to bind the node problem detector server. Use 0 to disable.")
	fs.StringVar(&npdo.ServerAddress, "address", "127.0.0.1",
		"The address to bind the node problem detector server.")
```
and 
```
	fs.IntVar(&npdo.PrometheusServerPort, "prometheus-port", 20257,
		"The port to bind the Prometheus scrape endpoint. Prometheus exporter is enabled by default at port 20257. Use 0 to disable.")
	fs.StringVar(&npdo.PrometheusServerAddress, "prometheus-address", "127.0.0.1",
		"The address to bind the Prometheus scrape endpoint.")
```